# HG changeset patch
# User Rik <rik@octave.org>
# Date 1520291900 28800
#      Mon Mar 05 15:18:20 2018 -0800
# Node ID 08a591a59a953e85bb38e4c5efb4d0e43316f695
# Parent  ae6679369a80fc73efe06c64bd4ff89b744232f4
plot3.m: Fix bug introduced in cset d55e100aebc9.

* plot3.m: Initialize possible output variable htmp at the top of the function.

diff --git a/scripts/plot/draw/plot3.m b/scripts/plot/draw/plot3.m
--- a/scripts/plot/draw/plot3.m
+++ b/scripts/plot/draw/plot3.m
@@ -119,16 +119,17 @@ function h = plot3 (varargin)
     x_set = 0;
     y_set = 0;
     z_set = 0;
     property_set = 0;
     fmt_set = 0;
     properties = {};
     tlgnd = {};
     hlgnd = [];
+    htmp = [];
     idx = 0;
 
     ## Gather arguments, decode format, and plot lines.
     arg = 0;
     while (arg++ < nargs)
       new = varargin{arg};
       new_cell = varargin(arg);
 
@@ -326,17 +327,16 @@ function h = plot3 (varargin)
       if (! size_equal (x, y, z))
         error ("plot3: X, Y, and Z must have the same shape");
       elseif (ndims (x) > 2)
         error ("plot3: X, Y, and Z must not have more than two dimensions");
       endif
 
       options = __default_plot_options__ ();
 
-      htmp = [];
       for i = 1 : columns (x)
         linestyle = options.linestyle;
         marker = options.marker;
         if (isempty (marker) && isempty (linestyle))
           [linestyle, marker] = __next_line_style__ ();
         endif
         color = options.color;
         if (isempty (color))
